### PR TITLE
Bluetooth: controller: Use nRFx functions for PPI reg

### PR DESCRIPTION
**Pull request by aescolar**
In the BLE controller, NRF radio HAL, for the PPI registers used for the SW TIFS.
To allow easier testing, replace direct use of registers from Nordic's nrfx MDK with accesses via its HAL inlined functions

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

### CODE REVIEW

**Review by cvinayak — COMMENTED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]
LGTM, but @cvinayak needs to approve

**Comment by carlescufi** [comment]
Closing/reopening due to CI hung

**Comment by aescolar** [comment]
@carlescufi : Thanks. But note that until the new nrfx hal PPI function make it in, CI will not pass. That's the reason for the DNM tag.

**Comment by aescolar** [comment]
@cvinayak : Do you think this changes can be approved?

File: subsys/bluetooth/controller/hal/nrf5/radio/radio.c [hunk 1]
```diff
@@ -359,13 +359,12 @@ static void sw_switch(u8_t dir, u8_t phy_curr, u8_t flags_curr, u8_t phy_next,
...
-	    HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_EVT;
-	HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_REGISTER_TASK =
-	    HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_TASK(sw_tifs_toggle);
+	nrf_ppi_channel_endpoint_setup(HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI,
+		HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_EVT,
+		HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_TASK(sw_tifs_toggle));
 
-	HAL_SW_SWITCH_RADIO_ENABLE_PPI_REGISTER_EVT(ppi) =
-		HAL_SW_SWITCH_RADIO_ENABLE_PPI_EVT(cc);
+	nrf_ppi_event_endpoint_setup(ppi,
```
Comment (anangl) [inline subsys/bluetooth/controller/hal/nrf5/radio/radio.c]: Why not do the same in line 411 and get rid of `HAL_SW_SWITCH_RADIO_ENABLE_PPI_REGISTER_EVT` completely?

Comment (aescolar) [reply subsys/bluetooth/controller/hal/nrf5/radio/radio.c]: I just didn't go thru the code meant only for CONFIG_BT_CTLR_PHY_CODED or CONFIG_SOC_NRF52840 (not supported in the simulated board yet).

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/controller/hal/nrf5/radio/radio.c (hunk 1 above)
Comment (carlescufi) [reply subsys/bluetooth/controller/hal/nrf5/radio/radio.c]: @aescolar do you mind doing that, for consistency?

Comment (aescolar) [reply subsys/bluetooth/controller/hal/nrf5/radio/radio.c]: If you don't mind lets leave it for a future PR. I have my hands quite full now, and this code is very tedious to change. I would not like to botch it.

Comment (carlescufi) [reply subsys/bluetooth/controller/hal/nrf5/radio/radio.c]: OK. I'll merge as-is, this is not a big deal at this point.

**Pull request merged**